zerver/tests/test_auth_backends.py

```diff
@@ -3960,10 +3962,10 @@ def test_django_to_ldap_username_with_email_search(self) -> None:
         self.assertEqual(self.backend.django_to_ldap_username("aaron@zulip.com"),
                          self.ldap_username("aaron"))
 
-        with mock.patch("zproject.backends.logging.warning") as mock_warn:
+        with self.assertLogs(level="WARNING") as m:
             with self.assertRaises(ZulipLDAPExceptionNoMatchingLDAPUser):
                 self.backend.django_to_ldap_username("shared_email@zulip.com")
-            mock_warn.assert_called_with("Multiple users with email %s found in LDAP.", "shared_email@zulip.com")
+        self.assertEqual(m.output, ["WARNING:root:Multiple users with email shared_email@zulip.com found in LDAP."])
```

Comment:
Though this works, I think it's probably important to use the % formatting as it makes it clear that, that string is generated by placing an email there and we won't easily miss it when reading tests.

Reply:
made this change in [commit](https://github.com/zulip/zulip/pull/15708/commits/4e553ae59068e4d9243b83410189c185a8ca20dd)